lib/ramble/ramble/workspace/workspace.py

```diff
@@ -1089,82 +1095,137 @@ def create_mirror(self, mirror_root):
-            self._experiment_script = all_experiments_file
+        # Check that the workspace log directory exists
+        if not os.path.exists(self.log_dir):
+            os.makedirs(self.log_dir)
 
-        experiment_set = self.build_experiment_set()
+        # Set up a log file for the overall pipeline, excluding experiments
+        dt = self._date_string()
+        inner_delim = '.'
+        pipeline_log_base = pipeline + inner_delim + dt
+        pipeline_log_file = pipeline_log_base + '.out'
+        pipeline_log_path = os.path.join(self.log_dir, pipeline_log_file)
+        pipeline_log_stream = open(pipeline_log_path, 'a+')
+
+        # Set up a directory to contain a log for each experiment
+        self.experiment_log_dir = os.path.join(self.log_dir, pipeline_log_base)
+        if not os.path.exists(self.experiment_log_dir):
+            os.makedirs(self.experiment_log_dir)
+
+        with self.logger(pipeline_log_stream, echo=self.dry_run):
+            with self.logger.force_echo():
+                tty.msg('Streaming details to log:')
+                tty.msg(f'  {pipeline_log_path}')
+                if self.dry_run:
+                    cprint('@*g{      -- DRY-RUN -- DRY-RUN -- DRY-RUN -- DRY-RUN -- DRY-RUN --}')
+
+            all_experiments_file = None
+            experiment_set = ramble.experiment_set.ExperimentSet(self)
+            self.software_environments = \
+                ramble.software_environments.SoftwareEnvironments(self)
+
+            if not self.is_concretized():
+                error_message = 'Cannot run %s in a ' % pipeline + \
+                                'non-conretized workspace\n' + \
+                                'Run `ramble workspace concretize` on this ' + \
+                                'workspace first.\n' + \
+                                'Then ensure its spack configuration is ' + \
+                                'properly configured.'
+                with self.logger.force_echo():
+                    tty.die(error_message)
+
+            workspace_success = {
+                namespace.success: ramble.config.config.get_config(namespace.success)
+            }
+            self.extract_success_criteria('workspace', workspace_success)
+
+            if pipeline == 'setup':
+                all_experiments_file = open(self.all_experiments_path, 'w+')
+                all_experiments_file.write('#!/bin/sh\n')
+                self._experiment_script = all_experiments_file
+
+            experiment_set = self.build_experiment_set()
+
+            experiment_count = len(experiment_set.experiments) + \
+                len(experiment_set.chained_experiments)
+            with self.logger.force_echo():
+                tty.msg(f'  Working on {experiment_count} experiments:')
+
+        for exp, app_inst, count in experiment_set.all_experiments():
+            # Set up an experiment logger and a log file for each experiment
+            exp_log_file = exp + '.out'
```

Comment:
The application instance should own this filename creation.

Maybe something like:
`app_inst.log_file_name`